src/ToolingSharedLibrary/Includes/VbsEnclaveABI/Shared/ConversionHelpers.h

```diff
@@ -394,5 +403,28 @@ namespace VbsEnclaveABI::Shared::Converters
 
         return target_struct;
     }
+
+    template<typename Src, typename Target>
+    inline void UpdateParameterValue(Src& src, Target& target)
+    {
+        if constexpr (!ArePtrTypes<Src, Target> || AreBothUniquePtrs<Src, Target>)
+        {
+             target = std::move(src);
+        }
+        else if constexpr (ArePtrTypes<Src, Target>)
+        {
+            // Note: source could be a unique ptr and target could be a raw ptr as is the case for
+            // pointer inout values.
+            if (src && target)
+            {
+                *target = std::move(*src);
+            }
+        }
+        else
+        {
+            static_assert(IsPtrType<Src> || IsPtrType<Target>,
+                   "When either src or target type is a pointer, the other type is expected to be a pointer type.");
```

Comment:
could this ever hit?

the expression,
    EXP = ArePtrTypes<Src, Target>

is either true or it isn't, and you're testing for both EXP and !EXP (lines 410 and 414), so how could the else (423) ever trigger?